Current master benchmark segfaults with uvector

With gcc 6.3 as compiler on Fedora 25.
```
➜  trash cd xtensor 
➜  xtensor git:(master) mkdir build
➜  xtensor git:(master) cd build 
➜  build git:(master) cmake ..
 -- The C compiler identification is GNU 6.3.1
-- The CXX compiler identification is GNU 6.3.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- xtensor v0.8.2
-- Configuring done
-- Generating done
-- Build files have been written to: /home/wolfv/Programs/trash/xtensor/build
➜  build git:(master) cmake .. -DBUILD_TESTS=ON
-- xtensor v0.8.2
-- Forcing tests build type to Release
-- Performing Test HAS_CPP14_FLAG
-- Performing Test HAS_CPP14_FLAG - Success
-- Found GTest: /usr/lib64/libgtest.so  
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Forcing tests build type to Release
-- Performing Test HAS_LTO_FLAG
-- Performing Test HAS_LTO_FLAG - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /home/wolfv/Programs/trash/xtensor/build
➜  build git:(master) make xbenchmark
Scanning dependencies of target benchmark_xtensor
[ 50%] Building CXX object benchmark/CMakeFiles/benchmark_xtensor.dir/main.cpp.o
[100%] Linking CXX executable benchmark_xtensor
[100%] Built target benchmark_xtensor
Scanning dependencies of target xbenchmark
Using steady_clock
period num: 1
period den: 1000000000
steady = true

***********************************
* AXPY 1D BENCHMARK : std::vector *
***********************************

xarray   iteration: 10.5839ms
xtensor  iteration: 6.92862ms
xarray  xiteration: 468.885ms
xtensor xiteration: 84.345ms
xarray    indexing: 23.529ms
xtensor   indexing: 9.16667ms

*** Error in `./benchmark_xtensor': free(): invalid next size (fast): 0x0000000001a34060 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x791fb)[0x7f5217b041fb]
/lib64/libc.so.6(+0x8288a)[0x7f5217b0d88a]
/lib64/libc.so.6(cfree+0x4c)[0x7f5217b112bc]
./benchmark_xtensor[0x408e77]
./benchmark_xtensor[0x401a59]
/lib64/libc.so.6(__libc_start_main+0xf1)[0x7f5217aab401]
./benchmark_xtensor(_start+0x2a)[0x401ada]
======= Memory map: ========
00400000-0040f000 r-xp 00000000 fd:03 3954011                            /home/wolfv/Programs/trash/xtensor/build/benchmark/benchmark_xtensor
0060e000-0060f000 r--p 0000e000 fd:03 3954011                            /home/wolfv/Programs/trash/xtensor/build/benchmark/benchmark_xtensor
0060f000-00610000 rw-p 0000f000 fd:03 3954011                            /home/wolfv/Programs/trash/xtensor/build/benchmark/benchmark_xtensor
01a1c000-01a4e000 rw-p 00000000 00:00 0                                  [heap]
7f5210000000-7f5210021000 rw-p 00000000 00:00 0 
7f5210021000-7f5214000000 ---p 00000000 00:00 0 
7f5217a8b000-7f5217c48000 r-xp 00000000 fd:02 2104949                    /usr/lib64/libc-2.24.so
7f5217c48000-7f5217e47000 ---p 001bd000 fd:02 2104949                    /usr/lib64/libc-2.24.so
7f5217e47000-7f5217e4b000 r--p 001bc000 fd:02 2104949                    /usr/lib64/libc-2.24.so
7f5217e4b000-7f5217e4d000 rw-p 001c0000 fd:02 2104949                    /usr/lib64/libc-2.24.so
7f5217e4d000-7f5217e51000 rw-p 00000000 00:00 0 
7f5217e51000-7f5217e67000 r-xp 00000000 fd:02 2106204                    /usr/lib64/libgcc_s-6.3.1-20161221.so.1
7f5217e67000-7f5218066000 ---p 00016000 fd:02 2106204                    /usr/lib64/libgcc_s-6.3.1-20161221.so.1
7f5218066000-7f5218067000 r--p 00015000 fd:02 2106204                    /usr/lib64/libgcc_s-6.3.1-20161221.so.1
7f5218067000-7f5218068000 rw-p 00016000 fd:02 2106204                    /usr/lib64/libgcc_s-6.3.1-20161221.so.1
7f5218068000-7f5218170000 r-xp 00000000 fd:02 2105597                    /usr/lib64/libm-2.24.so
7f5218170000-7f521836f000 ---p 00108000 fd:02 2105597                    /usr/lib64/libm-2.24.so
7f521836f000-7f5218370000 r--p 00107000 fd:02 2105597                    /usr/lib64/libm-2.24.so
7f5218370000-7f5218371000 rw-p 00108000 fd:02 2105597                    /usr/lib64/libm-2.24.so
7f5218371000-7f52184e9000 r-xp 00000000 fd:02 2104636                    /usr/lib64/libstdc++.so.6.0.22
7f52184e9000-7f52186e9000 ---p 00178000 fd:02 2104636                    /usr/lib64/libstdc++.so.6.0.22
7f52186e9000-7f52186f3000 r--p 00178000 fd:02 2104636                    /usr/lib64/libstdc++.so.6.0.22
7f52186f3000-7f52186f5000 rw-p 00182000 fd:02 2104636                    /usr/lib64/libstdc++.so.6.0.22
7f52186f5000-7f52186f9000 rw-p 00000000 00:00 0 
7f52186f9000-7f521871e000 r-xp 00000000 fd:02 2104671                    /usr/lib64/ld-2.24.so
7f52188ce000-7f52188d2000 rw-p 00000000 00:00 0 
7f521891b000-7f521891e000 rw-p 00000000 00:00 0 
7f521891e000-7f521891f000 r--p 00025000 fd:02 2104671                    /usr/lib64/ld-2.24.so
7f521891f000-7f5218920000 rw-p 00026000 fd:02 2104671                    /usr/lib64/ld-2.24.so
7f5218920000-7f5218921000 rw-p 00000000 00:00 0 
7ffce43e7000-7ffce4409000 rw-p 00000000 00:00 0                          [stack]
7ffce4587000-7ffce4589000 r--p 00000000 00:00 0                          [vvar]
7ffce4589000-7ffce458b000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
/bin/sh: line 1: 13390 Aborted                 (core dumped) ./benchmark_xtensor
benchmark/CMakeFiles/xbenchmark.dir/build.make:57: recipe for target 'benchmark/CMakeFiles/xbenchmark' failed
make[3]: *** [benchmark/CMakeFiles/xbenchmark] Error 134
CMakeFiles/Makefile2:206: recipe for target 'benchmark/CMakeFiles/xbenchmark.dir/all' failed
make[2]: *** [benchmark/CMakeFiles/xbenchmark.dir/all] Error 2
CMakeFiles/Makefile2:213: recipe for target 'benchmark/CMakeFiles/xbenchmark.dir/rule' failed
make[1]: *** [benchmark/CMakeFiles/xbenchmark.dir/rule] Error 2
Makefile:201: recipe for target 'xbenchmark' failed
make: *** [xbenchmark] Error 2
```

--- a/include/xtensor/xcontainer.hpp
+++ b/include/xtensor/xcontainer.hpp
@@ -226,7 +226,7 @@ namespace xt
 
     protected:
 
-        xstrided_container() = default;
+        xstrided_container() noexcept;
         ~xstrided_container() = default;
 
         xstrided_container(const xstrided_container&) = default;
@@ -714,6 +714,13 @@ namespace xt
     /*************************************
      * xstrided_container implementation *
      *************************************/
+
+    template <class D>
+    inline xstrided_container<D>::xstrided_container() noexcept
+        : base_type()
+    {
+        m_shape = make_sequence<inner_shape_type>(base_type::dimension(), 1);
+    }
 
     template <class D>
     inline xstrided_container<D>::xstrided_container(inner_shape_type&& shape, inner_strides_type&& strides) noexcept

--- a/include/xtensor/xstorage.hpp
+++ b/include/xtensor/xstorage.hpp
@@ -169,11 +169,14 @@ namespace xt
         {
             using pointer = typename A::pointer;
             using value_type = typename A::value_type;
-            if (!std::is_trivially_default_constructible<value_type>::value)
+            if (ptr != nullptr)
             {
-                for (pointer p = ptr; p != ptr + size; ++p)
+                if (!std::is_trivially_default_constructible<value_type>::value)
                 {
-                    alloc.destroy(p);
+                    for (pointer p = ptr; p != ptr + size; ++p)
+                    {
+                        alloc.destroy(p);
+                    }
                 }
                 alloc.deallocate(ptr, size);
             }
@@ -197,12 +200,12 @@ namespace xt
     inline void uvector<T, A>::resize_impl(size_type new_size)
     {
         size_type old_size = size();
+        pointer old_begin = p_begin;
         if (new_size != old_size)
         {
-            pointer tmp = detail::safe_init_allocate(m_allocator, new_size);
-            std::swap(tmp, p_begin);
+            p_begin = detail::safe_init_allocate(m_allocator, new_size);
             p_end = p_begin + new_size;
-            detail::safe_destroy_deallocate(m_allocator, tmp, old_size);
+            detail::safe_destroy_deallocate(m_allocator, old_begin, old_size);
         }
     }
 
@@ -261,6 +264,7 @@ namespace xt
     {
         detail::safe_destroy_deallocate(m_allocator, p_begin, size());
         p_begin = nullptr;
+        p_end = nullptr;
     }
 
     template <class T, class A>

